### PR TITLE
Refactor install_sct

### PR DESCRIPTION
**Pull request by gmotzespina**
# Changes

- `die` function added: Throws an error in red color, easier to spot when something happened.
- `check_requirements` function added: Checks if either `wget` or `curl` and `git` are installed before starting with the rest of the script. This saves the user time.
- Check after `pip install`: Throws an error if something wrong happened.
- Small refactor on functions. Keyword `function` added before each function name.

### CODE REVIEW

File: install_sct [hunk 1]
```diff
@@ -41,9 +41,30 @@ DATA_DIR="data"
...
+    echo -e "\n\033[0;31m$1\033[0m\n"
+    exit 1
+}
+
+function check_requirements {
+
+  if [[ ! $( which curl ) && ! $( which wget ) ]] ; then
+      die "You need to install \"curl\" OR \"wget\" before you can install the SPINALCORDTOOLBOX \n
+run \n
+> sudo apt-get install wget \n
```
Comment (jcohenadad) [inline install_sct]: the message needs to be distros-agnostic otherwise this command with fail on redhat, osx, etc.

Comment (gmotzespina) [reply install_sct]: You're completely right, I actually left it as it was, but I'll change it. Nice catch !

**Comment by jcohenadad** [comment]
@gmotzespina there is a "(WIP)" prefix in the title of this PR-- is this a mistake and the PR is ready for review, or are you still working on it?

**Comment by gmotzespina** [comment]
@jcohenadad sorry, my mistake, it is ready for review.

File: install_sct [hunk 2]
```diff
@@ -41,9 +41,27 @@ DATA_DIR="data"
 PYTHON_DIR="python"
 BIN_DIR="bin"
 
+function die {
+    echo -e "\n\033[0;31m$1\033[0m\n"
```
Comment (jcohenadad) [inline install_sct]: in the rest of this script, the convention is two chars for tabulation, so i would stick to 2 chars everywhere

File: install_sct [hunk 3]
```diff
@@ -41,9 +41,27 @@ DATA_DIR="data"
...
+    exit 1
+}
+
+function check_requirements {
+
+  if [[ ! $( which curl ) && ! $( which wget ) ]] ; then
+      die "Install \"curl\" OR \"wget\" before you can install the SPINALCORDTOOLBOX"
```
Comment (jcohenadad) [inline install_sct]: same comment about 2chars for indentation

File: install_sct [hunk 4]
```diff
@@ -41,9 +41,27 @@ DATA_DIR="data"
...
+  fi
+
+}
+
 # Force a clean exit
-finish ()
-{
+function finish {
     # Catch the last return code
     value=$?
```
Comment (jcohenadad) [inline install_sct]: same comment about indentation

File: install_sct [hunk 5]
```diff
@@ -111,46 +127,31 @@ edit_bashrc ()
 }
 
 # Download from URL using curl/wget
-download ()
-{
+function download {
   # Use curl or wget to download goodies
     e_status=0
```
Comment (jcohenadad) [inline install_sct]: indentation

File: install_sct [hunk 6]
```diff
@@ -209,9 +212,8 @@ if [[ -e "version.txt" ]]; then
     SCT_VERSION=${SCT_VERSION}_mpi
   fi
 else
-  echo "ERROR: version.txt not found."
-  echo "The install_sct script must be execute from the source directory"
-  exit 1
+  die "ERROR: version.txt not found. \n
+The install_sct script must be execute from the source directory"
```
Comment (jcohenadad) [inline install_sct]: must be execute --> must be executed

Comment (gmotzespina) [reply install_sct]: Ups, I missed that one. Thanks!

File: install_sct (hunk 5 above)
Comment (gmotzespina) [reply install_sct]: Sorry, changing it!

**Comment by gmotzespina** [comment]
@jcohenadad I've changed the identation for the file, could you please review it again?

**Pull request merged**